crates/evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -201,6 +202,25 @@ pub struct Cheatcodes {
     /// Breakpoints supplied by the `vm.breakpoint("<char>")` cheatcode
     /// char -> pc
     pub breakpoints: Breakpoints,
+
+    // Track if cool cheatcode was called on each address
+    pub cool: HashMap<Address, CoolState>,
+    // Track if an addresses storage slot is cool or not
+    pub cool_storage: HashMap<Address, HashMap<U256, CoolStorageState>>,
+    // How much gas to add in the next step (op) to have a cool storage slot
+    pub cool_additional_gas_next_op: u64,
+}
+
+#[derive(Clone, Debug)]
+pub enum CoolState {
+    CalledAndAccessed,
+    Called,
+}
+
+#[derive(Clone, Debug)]
+pub enum CoolStorageState {
+    Warm,
+    WarmAndSSTORED,
```

Comment:
FWIW I can't tell what this enum is supposed to represent from the names here alone, so we probably need to change the names and agree with adding docs. `CoolStorageState::Warm` and `CoolStorageState::WarmAndSSTORED` feel like conflicting names, e.g. how is the state cool and warm? Also maybe `WarmWithSSTORE` or `WarmWithSstore` would be easier to read since `SSTORED` isn't a usage I've seen

Reply:
right, i didn't know what to name the hashmap itself so it's `coolstoragestate` but it's more the storage state that is tracked when calling `vm.cool`, but it's tracking whether it's warm, or warm and the storage store was set, or that it's cool. Maybe very confusing but if it's empty it means it's cool (because each vm.cool resets the whole hashmap to be empty)

so it's either empty value (none), or warm (the sload makes it warm), or warm and sstore was called

I suppose this means cool, sloaded, or sstored, and each of those should change the gas increase